File: app/src/main/java/com/cyberegylet/antiDupeGallery/models/ImageFile.java
package com.cyberegylet.antiDupeGallery.models;

import com.cyberegylet.antiDupeGallery.backend.FileManager;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.nio.file.attribute.BasicFileAttributes;
import java.util.Objects;

public class ImageFile
{
	private File file;
	private String name;
	private long size;
	private long modifiedDate;
	private long creationDate;
	private boolean isHidden;
	private String mime;
	private long id;

	public ImageFile(File file) { this(file, "*/*", -1); }

	public ImageFile(File file, String mime, long id)
	{
		String stringPath = file.getPath();
		this.file = file;
		this.name = file.getName();
		this.isHidden = Objects.requireNonNull(name).charAt(0) == '.';
		this.mime = mime;
		this.id = id;
		try
		{
			BasicFileAttributes attr = Files.readAttributes(Paths.get(stringPath), BasicFileAttributes.class);
			size = FileManager.getSize(file);
			modifiedDate = attr.lastModifiedTime().toMillis();
			creationDate = attr.creationTime().toMillis();
		}
		catch (IOException e)
		{
			throw new RuntimeException(e);
		}
	}

	public File getFile() { return file; }

	public String getPath() { return file.getPath(); }

	public String getName() { return name; }

	public long getSize() { return size; }

	public long getModifiedDate() { return modifiedDate; }

	public long getCreationDate() { return creationDate; }

	public boolean isHidden() { return isHidden; }

	public String getMime() { return mime; }

	public long getId() { return id; }
}

File: app/src/main/java/com/cyberegylet/antiDupeGallery/models/ImageFile.kt
package com.cyberegylet.antiDupeGallery.models

import com.cyberegylet.antiDupeGallery.backend.FileManager
import java.io.File
import java.io.IOException
import java.nio.file.Files
import java.nio.file.Paths
import java.nio.file.attribute.BasicFileAttributes

class ImageFile @JvmOverloads constructor(file: File, val mime: String = "*/*", id: Long = -1) : FileEntry(file, id)
{
	val mimeEnum: FileManager.Mimes.Type = when
	{
		mime[0] == 'i' -> FileManager.Mimes.Type.IMAGE
		mime[0] == 'v' -> FileManager.Mimes.Type.VIDEO
		else -> FileManager.Mimes.Type.UNKNOWN
	}

	override fun mySetFile()
	{
		super.mySetFile()
		try
		{
			val attr = Files.readAttributes(Paths.get(file.path), BasicFileAttributes::class.java)
			size = FileManager.getSize(file)
			modifiedDate = attr.lastModifiedTime().toMillis()
			creationDate = attr.creationTime().toMillis()
		}
		catch (e: IOException)
		{
			throw RuntimeException(e)
		}
	}
}